src/Interpreters/Cache/FileCache.cpp

```diff
@@ -178,16 +224,7 @@ void FileCache::initialize()
 
     try
     {
-        if (fs::exists(getBasePath()))
-        {
-            loadMetadata();
-        }
-        else
-        {
-            fs::create_directories(getBasePath());
-        }
-
-        status_file = make_unique<StatusFile>(fs::path(getBasePath()) / "status", StatusFile::write_full_info);
+        loadMetadata();
```

Comment:
We should not call loadMetadata() if base path did not exist, when we called initialize(). Because we will start background threads in vain. So we need to not call initializeImpl() from initialize() in lines 208-216 if we saw earlier that base path did not exist

Reply:
Done this